aws-s3/s3.ml

```diff
@@ -231,7 +231,13 @@ module Make(Io : Types.Io) = struct
   type range = { first: int option; last: int option }
 
   type nonrec 'a result = ('a, error) result Deferred.t
-  type 'a command = ?credentials:Credentials.t -> ?connect_timeout_ms:int -> endpoint:Region.endpoint -> 'a
+  type 'a command = ?credentials:Credentials.t -> ?connect_timeout_ms:int -> ?requester_pays:bool -> endpoint:Region.endpoint -> 'a
+
+  (* conditionally add a header indicating caller's willingness to pay
+     for AWS data transfer costs *)
+  let add_request_payer headers = function
```

Comment:
As I understand the API, this header must be present on calls to buckets where the requested acknowledges that the requester will be charged for the request. 

The name `request_payer` seems not to reflect this. I suggest renaming this to `confirm_requester_pays`, in line with the documentation.

The function here acually only adds the header if the second argument is true, so it should be named `maybe_...`. Also flipping the two arguments makes it easier to chain calls using the pipe (`|>`) operator.